internal/bminventory/inventory.go

```diff
@@ -3297,3 +3316,23 @@ func (b *bareMetalInventory) getOpenshiftVersionFromOCP(log logrus.FieldLogger)
 	splits := strings.Split(openshiftVersion, ".")
 	return splits[0] + "." + splits[1], nil
 }
+
+func (b bareMetalInventory) PermanentlyDeleteUnregisteredClustersAndHosts() {
```

Comment:
Suggesting to check leader here, there is no need to iterate over all the hosts and cluster if you are not the leader.